src/main/java/com/aws/greengrass/deployment/IotJobsHelper.java

```diff
@@ -412,6 +461,7 @@ public void requestNextPendingJobDocument() {
      */
     public void subscribeToJobsTopics() {
```

Comment:
does this properly handle getting interrupted? This method doesn't throw interrupted exception

Reply:
Individual subscribe methods inside it handle getting interrupted.

Reply:
But this method doesn't throw, so how is it being handled?

Reply:
Those methods absorb the exception and break out of their loop. We can bubble up the exception upto this method.